File: Scripts/server/function-composition/go/function-composition.go
package main

import "math"

type ffType func(float64) float64

func compose(f, g ffType) ffType {
    return func(x float64) float64 {
        return f(g(x))
    }
}

func executeTask(i int) int {

        sin_asin := compose(math.Sin, math.Asin)
        sin_asin((float64)(i))
	return i + 1
}

func main() { 
    	r := int(1)
    	for i := 0; i < 1000000000; i++ {
    		r = executeTask(i + r)
	}
    _ = r
}

File: Scripts/server/function-composition/php/function-composition.php
<?php
function compose($f, $g) {
  return function($x) use ($f, $g) { return $f($g($x)); };
}

$r = 1;

function executeTask($i) {

	$sin_asin = compose('sin', 'asin');
	return $sin_asin($i);
	
}

for ($i = 0 ; $i < 1000000000; $i += 1) {
	$r = executeTask($i);
}

?>
